src/DynamicRedirect.js

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2019 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+const { URL } = require('url');
+const { fetch } = require('@adobe/helix-fetch').context({
+  httpsProtocols:
+  /* istanbul ignore next */
+    process.env.HELIX_FETCH_FORCE_HTTP1 ? ['http1'] : ['http2', 'http1'],
+});
+
+const DEFAULT_TYPE = 'permanent';
+const FROM_NAMES = ['from', 'src', 'source', 'origin'];
+const TO_NAMES = ['to', 'target', 'dest', 'destination'];
+const TYPE_NAMES = ['type', 'kind'];
+
+function getProp(entry, names) {
+  const pair = Object.entries(entry).find(([key]) => names.includes(key.toLowerCase()));
+  return pair ? pair[1] : null;
+}
+
+function clean(entry) {
+  return {
+    from: getProp(entry, FROM_NAMES),
+    to: getProp(entry, TO_NAMES),
+    type: getProp(entry, TYPE_NAMES),
+  };
+}
+
+class DynamicRedirect {
+  constructor(src, logger) {
+    this._src = src;
+    this._data = null;
+    this._logger = logger;
+  }
+
+  async match(path) {
+    if (!this._data) {
+      try {
+        // eslint-disable-next-line no-underscore-dangle
+        const namespace = process.env.__OW_NAMESPACE || 'helix';
+        const url = new URL(`https://adobeioruntime.net/api/v1/web/${namespace}/helix-services/data-embed@v1`);
+        url.searchParams.append('src', this._src);
+        const res = await fetch(url.href);
+        this._data = (await res.json()).map(clean);
```

Comment:
you should probably check `res.ok`. the request could return a 404 with a json body....

Reply:
bf0845d checks for `ok`